In [ ]:
import warnings
warnings.filterwarnings('ignore')

In [ ]:
from scipy import stats

## One Sample T -Test (Parametric testing, NORMAL)

In [ ]:
ban_mar_age = [28,31,32,34,30,28,29,35,32,30]
print("Skewness:", stats.skew(ban_mar_age))
print("Kurtosis:", stats.kurtosis(ban_mar_age))

Skewness: 0.385594595949688
Kurtosis: -0.9401924494656089


H0 : Marriage age of IT MALE in Bangalore is 30 <br>
Ha : Marriage age of IT MALE in Bangalore is NOT 30 <br>
Hypothesized Mean : 30

In [ ]:
stats.ttest_1samp(ban_mar_age,30)

Ttest_1sampResult(statistic=1.196753959380426, pvalue=0.26197483832274515)

P = 26.2%, is NOT < 5%, we fail to Reject H0,ACCEPT H0 <br>

## Alternatives we can conclude using T value,  
Tvalue > Tcritical, We reject H0,   Tcritical from lookup table


## Two Sample Independent T-Test
H0: Ban and Che Marriage IT Male ages are SAME <br>
Ha: Ban and Che Marriage IT Male ages are NOT SAME 

In [ ]:
ban_mar_age = [28,31,32,34,30,28,29,35,32,30]
che_mar_age = [28,26,33,28,26,25,27,28,31,28]

In [ ]:
stats.ttest_ind(ban_mar_age,che_mar_age)

Ttest_indResult(statistic=2.71213663579019, pvalue=0.014278632125045938)

P = 1.4%, < 5%, We Reject the H0, ACCEPT Ha <br>
Ha: Ban and Che Marriage IT Male ages are NOT SAME<br>
Tvalue = 2.7, positive, So Ban Mar age are higher than Chennai 

## Two Sample Relational T-Test

In [ ]:
before_rb = [35,45,30,50,60] # response time centisecs
after_rb = [25,35,25,30,40] # response time centisecs

H0: Response times before and after RB are SAME <br>
Ha: Response times before and after RB are NOT SAME

In [ ]:
stats.ttest_rel(before_rb,after_rb)

Ttest_relResult(statistic=4.333333333333333, pvalue=0.012317352470240385)

P =1.2%, < 5%, we REJECT H0, ACCEPT Ha <br>
Ha: Response times before and after RB are NOT SAME <br>
Tvalue = 4.3, is positive, Before are higher than after<br>
RB gives you wings

# Analysis of Variance (ANOVA), F-Test

In [ ]:
lemon = [36,12,46,10,15,20]
coffee = [33,10,38,12,13,19]
rb = [31,9,39,13,10,15]

H0: All drinks response times are SAME <br>
Ha: All drinks response times are NOT SAME 

In [ ]:
stats.f_oneway(lemon,coffee,rb)

F_onewayResult(statistic=0.12208729898260585, pvalue=0.8859415816548135)

P = 88.5, NOT < 5%,  We ACCEPT H0 <br>
H0: All drinks response times are SAME, which mean drinks have not EFFECT

# Colleration

In [ ]:
discounts = [10,20,30,50,60]
sales = [10,12,15,17,17]
profit = [5,5,4,3,1]

In [ ]:
stats.pearsonr(discounts,sales)

(0.9522588572397535, 0.012431924047055558)

In [ ]:
stats.pearsonr(discounts,profit)

(-0.9510441892119876, 0.012906993663617769)